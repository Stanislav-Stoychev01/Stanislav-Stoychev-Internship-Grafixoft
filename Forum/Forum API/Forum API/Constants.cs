namespace Forum_API
{
    public static class Constants
    {
        public const string defaultRoleDescription = "default user";
        public const string dateFormat = "yyyy - MM - dd HH:mm:ss";
    }
}
