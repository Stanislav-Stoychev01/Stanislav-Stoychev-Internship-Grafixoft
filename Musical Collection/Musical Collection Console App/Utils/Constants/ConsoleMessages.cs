using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Musical_Collection_Console_App.Constants
{
    public class ConsoleMessages
    {
        public const string SuccessfulRegister = "You have registered succesfully";
        public const string SuccessfulLogin = "You have logged in succesfully";
        public const string UnsuccessfulLogin = "Invalid credentials";
        public const string InvalidAction = "The listener must be logged in first";
    }
}
