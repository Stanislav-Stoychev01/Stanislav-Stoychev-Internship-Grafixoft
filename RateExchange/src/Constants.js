export const currencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN'];

export const defaultCurrency = 'USD';


export const columnCount = 3;
export const firstColumn = 'First Column';
export const secondColumn = 'Second Column';
export const thirdColumn = 'Third Column';

export const firstColumnIndex = 0;
export const secondColumnIndex = 1;
export const thirdColumnIndex = 2;

export const key = 0;
export const value = 1;

export const dateString = 'date';
export const latest = 'latest';

export const inputLabel = "Input Currency Name:";
export const exchangeRateLabel = "Exchange Rates from";
export const longestSequenceMessage = "The longest sequence has length of ";

export const tableColumnNames = ['Conversion Rates below 1', 
'Conversion Rates between 1 and 1,5', 'Conversion Rates above 1,5'];

export const LoadingData = 'The data is currenly loading';