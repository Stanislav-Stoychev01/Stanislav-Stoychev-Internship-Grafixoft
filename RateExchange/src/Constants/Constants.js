export const CurrencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN']
export const DefaultCurrency = 'USD'
