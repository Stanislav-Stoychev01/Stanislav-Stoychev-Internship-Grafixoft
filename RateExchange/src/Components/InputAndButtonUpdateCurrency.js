import "../CSS/ButtonUpdateState.css";
import * as Constants from "../Constants";
import { useEffect, useRef, useState } from "react";
import * as LongestSequence from "../Utils/LongestSequence";
import { getCurrentDay } from "../Utils/Date"

function InputAndButtonUpdateCurrency(props)
{
    const inputRef = useRef();
    const [longestSequence, setLongestSequence] = useState();

    useEffect(() => {
        
        if(props.currency) {
            
            const longestSequence = LongestSequence.longestSequence(props.currency, localStorage);
            setLongestSequence(longestSequence);
        }

    }, [props.currency, props.areValuesUpdated])

    const onUpdateButtonClick = () => {

        const upperCaseCurrency = inputRef.current.value.toUpperCase();

        if (Constants.currencyList.includes(upperCaseCurrency)) {

            const currentDate = getCurrentDay();
            props.onUpdateButtonClick !== undefined &&
            props.onUpdateButtonClick(inputRef.current.value.toLowerCase(),
            currentDate);
        }

        else {
            alert("Invalid Currency name!");
        }

        inputRef.current.value = "";
    }

    const LabelAndInput = () => {
        return (
            <div className="input-container">
                <label className="label-input">
                    {props.label}
                </label>
                <input className="currency-input" ref={inputRef}></input>
            </div>
        )
    }

    const UpdateCurrencyButton = () => {
        return (
            <div className="button-container">
                {
                    props.areValuesUpdated === false &&
                    <div className="button-container">
                        <button className="update-button"
                            onClick={onUpdateButtonClick}>
                            { props.buttonText }
                        </button>
                    </div>
                }
            </div> 
        )
    }

    const LongestSequenceText = () => {
        return (
            <div className="longest-sequence-container">
                {
                    props.areValuesUpdated === true 
                    && <label className="longest-sequence">
                        {props.longestSequenceMessage} {longestSequence}
                    </label>
                }
            </div>
        )
    }
    
    return (
        <div className="update-state-container">                  
            <LabelAndInput />
            <LongestSequenceText />
            <UpdateCurrencyButton />
        </div>
    )
}

export default InputAndButtonUpdateCurrency;