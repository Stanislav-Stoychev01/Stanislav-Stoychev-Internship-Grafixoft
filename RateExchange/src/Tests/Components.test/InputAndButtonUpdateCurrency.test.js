import InputAndButtonUpdateCurrency from "../../Components/InputAndButtonUpdateCurrency";
import * as Constants from "../../Constants";
import Enzyme from 'enzyme';
import toJson from "enzyme-to-json";
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

Enzyme.configure({ adapter: new Adapter() });

it("shows input and update currency button when not all values are updated", () => {

    const currency = "usd";
    const areValuesUpdated = false;
    const { inputLabel, longestSequenceMessage, updateButtonName} = Constants;
    
    const props = {
        currency: currency,
        areValuesUpdated: areValuesUpdated,
        label: inputLabel,
        longestSequenceMessage: longestSequenceMessage,
        buttonText: updateButtonName
    };

    const wrapper = mount(<InputAndButtonUpdateCurrency {...props}/>);

    const wrapperToJson = toJson(wrapper);

    expect(wrapperToJson).toMatchSnapshot();
});

it("shows input and update currency button when not all values are updated", () => {

    const currency = "usd";
    const areValuesUpdated = true;
    const { inputLabel, longestSequenceMessage, updateButtonName } = Constants;

    const props = {
        currency: currency,
        areValuesUpdated: areValuesUpdated,
        label: inputLabel,
        longestSequenceMessage: longestSequenceMessage,
        buttonText: updateButtonName
    };

    const wrapper = mount(<InputAndButtonUpdateCurrency {...props} />);

    const wrapperToJson = toJson(wrapper);

    expect(wrapperToJson).toMatchSnapshot();
});