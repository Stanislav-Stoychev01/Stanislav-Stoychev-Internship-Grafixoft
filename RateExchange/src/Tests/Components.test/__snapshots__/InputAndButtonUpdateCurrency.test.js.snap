// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows input and update currency button when not all values are updated 1`] = `
<InputAndButtonUpdateCurrency
  areValuesUpdated={false}
  buttonText="Update Currency"
  currency="usd"
  label="Input Currency Name:"
  longestSequenceMessage="The longest sequence has length of "
>
  <div
    className="update-state-container"
  >
    <LabelAndInput>
      <div
        className="input-container"
      >
        <label
          className="label-input"
        >
          Input Currency Name:
        </label>
        <input
          className="currency-input"
        />
      </div>
    </LabelAndInput>
    <LongestSequenceText>
      <div
        className="longest-sequence-container"
      />
    </LongestSequenceText>
    <UpdateCurrencyButton>
      <div
        className="button-container"
      >
        <div
          className="button-container"
        >
          <button
            className="update-button"
            onClick={[Function]}
          >
            Update Currency
          </button>
        </div>
      </div>
    </UpdateCurrencyButton>
  </div>
</InputAndButtonUpdateCurrency>
`;

exports[`shows input and update currency button when not all values are updated 2`] = `
<InputAndButtonUpdateCurrency
  areValuesUpdated={true}
  buttonText="Update Currency"
  currency="usd"
  label="Input Currency Name:"
  longestSequenceMessage="The longest sequence has length of "
>
  <div
    className="update-state-container"
  >
    <LabelAndInput>
      <div
        className="input-container"
      >
        <label
          className="label-input"
        >
          Input Currency Name:
        </label>
        <input
          className="currency-input"
        />
      </div>
    </LabelAndInput>
    <LongestSequenceText>
      <div
        className="longest-sequence-container"
      >
        <label
          className="longest-sequence"
        >
          The longest sequence has length of 
           
        </label>
      </div>
    </LongestSequenceText>
    <UpdateCurrencyButton>
      <div
        className="button-container"
      />
    </UpdateCurrencyButton>
  </div>
</InputAndButtonUpdateCurrency>
`;
